kconfig-hardened-check.py

```diff
@@ -17,101 +17,123 @@
-    opt_list.append([Opt('SECURITY_SELINUX_DISABLE',    'is not set', 'ubuntu18', 'security_policy'), ''])
-
-    opt_list.append([Opt('ACPI_CUSTOM_METHOD',   'is not set', 'ubuntu18', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('COMPAT_BRK',           'is not set', 'ubuntu18', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('DEVKMEM',              'is not set', 'ubuntu18', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('COMPAT_VDSO',          'is not set', 'ubuntu18', 'cut_attack_surface'), ''])
-
-    opt_list.append([Opt('LEGACY_VSYSCALL_NONE', 'y', 'kspp', 'cut_attack_surface'), '']) # 'vsyscall=none'
-    opt_list.append([Opt('BINFMT_MISC',          'is not set', 'kspp', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('INET_DIAG',            'is not set', 'kspp', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('KEXEC',                'is not set', 'kspp', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('PROC_KCORE',           'is not set', 'kspp', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('LEGACY_PTYS',          'is not set', 'kspp', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('IA32_EMULATION',       'is not set', 'kspp', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('X86_X32',              'is not set', 'kspp', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('MODIFY_LDT_SYSCALL',   'is not set', 'kspp', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('KEXEC_FILE',           'is not set', 'my', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('LIVEPATCH',            'is not set', 'my', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('USER_NS',              'is not set', 'my', 'cut_attack_surface'), '']) # user.max_user_namespaces=0
-    opt_list.append([Opt('IP_DCCP',              'is not set', 'my', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('IP_SCTP',              'is not set', 'my', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('FTRACE',               'is not set', 'my', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('KPROBES',              'is not set', 'my', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('PROFILING',            'is not set', 'my', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('UPROBES',              'is not set', 'my', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('BPF_JIT',              'is not set', 'my', 'cut_attack_surface'), ''])
-    opt_list.append([Opt('BPF_SYSCALL',          'is not set', 'my', 'cut_attack_surface'), ''])
-
-    opt_list.append([Opt('LKDTM',    'm', 'my', 'feature_test'), ''])
-
-
-def print_opt_list():
+opt_checks = OrderedDict()
+
+
+class Opt(object):
+    def __init__(self, name, state, decision, reason):
+        self.name = name
+        self.state = state
+        self.decision = decision
+        self.reason = reason
+
+    def check(self, options):
+        global error_count
+        state = get_option(options, self.name)
```

Comment:
1. How about calling this function get_option_state()?
2. It took me some time to understand that 'options' here is a dictionary which we have created from the config file.
 - Would you mind to add a comment?
 - What do you think about calling it 'parsed_options' to avoid mixing with the dictionary with Opt objects.

By the way, ideally this class should be called something like OptCheck, right?

Reply:
sounds sane to me, gonna rename the things

Reply:
ok!